source/core/KeyboardEvent.cs

```diff
@@ -0,0 +1,22 @@
+using System.Windows.Forms;
+
+namespace SHVDN
+{
+    public struct KeyboardEvent
```

Comment:
We could add `readonly` so code readers won't think `KeyboardEvent` structs can be changed and the C# compiler won't defensive-copy ones when passing by reference (or calling a mutable method). `System.Tuple` is designed to be immutable, too.